### PR TITLE
Fix handling of redshift setting in output of specviz.get_spectra

### LINKED ISSUE #454: Applying slider redshift to output spectra changes spectral axis values
This is the intended behavior for updating the redshift value on a `specutils.Spectrum1D` object, but I think it's not what we want to happen when the Specviz `get_spectra` helper class is used. My understanding of what we really want is to just set the redshift parameter in the output `Spectrum1D` but leave the spectral axis unchanged, so the user could potentially use that redshift to e.g. calculate rest values if desired. 

I think the specutils treatments is actually, even worse, the direct opposite of what we want here: specutils is treating the original wavelengths as the rest values and converting to what they would be at the set redshift, but in our case the original values are redshifted and we would want to convert them back to rest.

I include a demonstration of the difference below.

Current behavior:
```
>>> flux = [1,2,3]* u.Jy
>>> sa = [300, 310, 320] * u.AA
>>> spec = Spectrum1D(flux=flux, spectral_axis=sa)
>>> spec
<Spectrum1D(flux=<Quantity [1., 2., 3.] Jy>, spectral_axis=<SpectralAxis [300., 310., 320.] Angstrom>)>
>>> spec.redshift
<Quantity 0.>
>>>
>>> spec.redshift = 0.56
>>> spec
<Spectrum1D(flux=<Quantity [1., 2., 3.] Jy>, spectral_axis=<SpectralAxis
   (observer to target:
      radial_velocity=125169.63763653308 km / s
      redshift=0.56)
  [468. , 483.6, 499.2] Angstrom>)>
```
What I think we want:
```
>>> spec2 = Spectrum1D(flux = spec.flux, spectral_axis = sa, redshift = 0.56)
>>> spec2
<Spectrum1D(flux=<Quantity [1., 2., 3.] Jy>, spectral_axis=<SpectralAxis
   (observer to target:
      radial_velocity=125169.63763653308 km / s
      redshift=0.56)
  [300., 310., 320.] Angstrom>)>
```

@eteq @camipacifici @PatrickOgle let me know your thoughts, this is a quick fix I just need to know to go ahead and do it.

### PR DESCRIPTION
**Pull request by rosteen**
Fixes #454 (assuming the POs agree with my assumptions there).

### CODE REVIEW

File: jdaviz/configs/specviz/helper.py [hunk 1]
```diff
@@ -39,17 +39,31 @@ def get_spectra(self, data_label=None, apply_slider_redshift="Warn"):
...
+            if data_label is not None:
+                spectra = {data_label: spectra}
+            for key in spectra.keys():
+                flux = spectra[key].flux
+                # This is a hack around inability to input separate redshift with
+                # a SpectralAxis instance in Spectrum1D
+                spaxis = spectra[key].spectral_axis.value * spectra[key].spectral_axis.unit
+                mask = spectra[key].mask
+                uncertainty = spectra[key].uncertainty
+                print(self._redshift)
```
Comment (javerbukh) [inline jdaviz/configs/specviz/helper.py:53]: I would suggest clarifying the print statement (i.e. "Redshift: ") or removing it altogether.

**Review by javerbukh — COMMENTED** [review]
I tried comparing the results of `get_spectra()` with `apply_slider_redshift` as True and as False - with a redshift of 0.765 - and the spectral axes were identical. Am I doing something wrong on my end?

File: jdaviz/configs/specviz/helper.py (hunk 1 above)
Comment (rosteen) [reply jdaviz/configs/specviz/helper.py]: Good catch, this was a leftover from debugging.

**Comment by rosteen** [comment]
@javerbukh Nothing wrong on your end, that's the intended functionality implemented by this PR. `apply_slider_redshift` should only set the `redshift` parameter of the output `Spectrum1D`, not also change the spectral axis as it was previously doing.

**Comment by javerbukh** [comment]
@rosteen Is there no situation where a user would want to apply the redshift to the spectral axis? If I were a first time user I would imagine that is what `apply_slider_redshift=True` would do. Am I wrong in thinking this @camipacifici @eteq ? If so, this PR looks good to me and I will come back and approve it.

**Comment by rosteen** [comment]
The distinction here (which is indeed a little confusing) is that changing the axis along with the redshift (as generally happens when you set the redshift in specutils, which is what I'm working around here) does the operation "show me what this spectrum would look like at the new redshift." Whereas the case of working with the Specviz redshift slider is "I know this spectrum is already redshifted but I don't know the redshift value, let me figure it out then attach that value to the spectrum." In the second case (our use in Specviz) the user could then use `spectrum.to_rest()` to see what the original data would look like at rest, or convert it to any other redshift.

**Review by javerbukh — APPROVED** [review]
@rosteen Ah, that makes sense, thank you for the explanation! This looks good to me then.

**Comment by rosteen** [comment]
For posterity: @camipacifici pointed out offline that my reference above to doing `spectrum.to_rest()` is incorrect. For a `Spectrum1D` you just set `spectrum.redshift=0` to do that, which caused the problem I'm solving here. `to_rest` is a method in the `SpectralCoord` class.

**Review by duytnguyendtn — APPROVED** [review]
Read through the thread and I *think* I understand the expected behavior. If so, this looks good!

Confirmed that setting the redshift does not change the actual data itself, but just sets the redshift value

**Pull request merged**